tests/test_type_inference/test_comprehension.py

```diff
@@ -0,0 +1,34 @@
+import astroid
+import nose
+from hypothesis import given, settings, assume
+from typing import Callable, Any
+import tests.custom_hypothesis_support as cs
+settings.load_profile("pyta")
+
+
+# def test_list_comprehension1():
+#     """Test Comprehension node visitor representing a comprehension expression."""
+#     program = f'a = [num for num in [1,2,3]]'
```

Comment:
If you don't have the simple one working then you shouldn't be working on the more complex one.